system/jlib/jthread.cpp

```diff
@@ -1017,13 +1026,11 @@ class CThreadPool: public CThreadPoolBase, implements IThreadPool, public CInter
                         throw MakeStringException(0, "No threads available in pool %s", poolname.get());
                     IWARNLOG("Pool limit exceeded for %s", poolname.get());
                 }
-                else
-                    timedout = false;
             }
             if (traceStartDelayPeriod)
             {
                 ++startsInPeriod;
-                if (timedout)
+                if (waited)
```

Comment:
Don't think it really matters, but if there's an accumulate delay, but as it crosses the 1 minute mark (the 'traceStartDelayPeriod' period), 'waited' isn't true, then it won't report until the next delayed thread.

Perhaps change to:
```
            if (traceStartDelayPeriod)
            {
                ++startsInPeriod;
                if (waited)
                    startDelayInPeriod += startTimer.elapsedCycles();
                if (startDelayInPeriod && (overAllTimer.elapsedCycles() >= queryOneSecCycles()*traceStartDelayPeriod)) // check avg. delay per minute
                {
....
                }
            }
```

Reply:
I agree, good find.  I would even say tho I want to know how many threads started even if there is no accumulated delay, so I don't think we need the extra check of if(startDelayInPeriod) to log the info.  See commit 2.